code/ZAS/Fire.dm

```diff
@@ -43,101 +83,151 @@ Attach to transfer valve and open. BOOM.
 /atom/proc/ashtype()
 	return /obj/effect/decal/cleanable/ash
 
-/atom/proc/getFireFuel()
-	return fire_fuel
+//this proc is called on every fire/process()
+//energy is taken from burning atoms and delivered to the fire at its current location
+//proc returns energy in MJ and oxygen consumed in mols
+/atom/proc/burnSolidFuel()
+	var/turf/T = get_turf(loc)
 
-/atom/proc/burnFireFuel(used_fuel_ratio, used_reactants_ratio)
-	fire_fuel -= (fire_fuel * used_fuel_ratio * used_reactants_ratio) //* 5
+	var/datum/gas_mixture/air = T.return_air()
+	var/oxy_ratio  = air.molar_density(GAS_OXYGEN)
+	var/pressure = air.return_pressure()
+	var/temperature = air.return_temperature()
+	var/delta_t
 
-	var/turf/T = get_turf(loc)
-	if(T)
-		T.hotspot_expose(autoignition_temperature, CELL_VOLUME, surfaces=1)
-	if(prob(8)) //8% chance of smoke creation per tick
-		var/datum/effect/system/smoke_spread/fire/smoke = new /datum/effect/system/smoke_spread()
-		smoke.set_up(4,0,T)
-		smoke.time_to_live = 60 SECONDS
-		smoke.start()
+	var/heat_out = 0 //MJ
+	var/oxy_used = 0 //mols
+	var/list/burn_products = list(heat_out,oxy_used)
```

Comment:
this proc will always return [0,0]
you'd need to make them objects for it to use references instead of copying